lib/std/Thread/Condition.zig

```diff
@@ -63,6 +74,17 @@ pub const WindowsCondition = struct {
         assert(rc != windows.FALSE);
     }
 
+    pub fn timedWait(cond: *WindowsCondition, mutex: *Mutex, timeout_ns: u64) error{TimedOut}!void {
+        const rc = windows.kernel32.SleepConditionVariableSRW(
+            &cond.cond,
+            &mutex.impl.srwlock,
+            @truncate(windows.DWORD, timeout_ns / std.time.ns_per_ms),
+            @as(windows.ULONG, 0),
+        );
+        if (rc == windows.FALSE and windows.kernel32.GetLastError() == windows.Win32Error.TIMEOUT) return error.TimedOut;
```

Comment:
Maybe assert that `GetLastError() == .TIMEOUT`?

Reply:
Seems like this is covered by `assert(rc != windows.FALSE);` on the next line? If GetLastError is TIMEOUT, the function returns with error.Timeout. Any other error error will lead to the assert triggering.

Reply:
You're right, didn't realize the assert right after. Not sure if this is still true with the timeout overflow checking however.